diff --git a/node_modules/@capacitor/ios/Capacitor/Capacitor/WebViewDelegationHandler.swift b/node_modules/@capacitor/ios/Capacitor/Capacitor/WebViewDelegationHandler.swift
index 4ec797b..62070d0 100644
--- a/node_modules/@capacitor/ios/Capacitor/Capacitor/WebViewDelegationHandler.swift
+++ b/node_modules/@capacitor/ios/Capacitor/Capacitor/WebViewDelegationHandler.swift
@@ -158,7 +158,20 @@ internal class WebViewDelegationHandler: NSObject, WKNavigationDelegate, WKUIDel
     }
 
     public func webViewWebContentProcessDidTerminate(_ webView: WKWebView) {
-        webView.reload()
+        guard let url = webView.url else {
+            webView.reload()
+            return
+        }
+        guard var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: true) else {
+            webView.reload()
+            return
+        }
+        urlComponents.queryItems = (urlComponents.queryItems ?? []) + [URLQueryItem(name: "crash", value: "true")]
+        guard let crashUrl = urlComponents.url else {
+            webView.reload()
+            return
+        }
+        webView.load(URLRequest(url: crashUrl))
     }
 
     // MARK: - WKScriptMessageHandler
